packages/app/src/components/TokenBalance.tsx

```diff
@@ -0,0 +1,70 @@
+'use client'
+import { useBalance, useReadContract } from 'wagmi'
+import { formatEther } from 'viem'
+import { toBigInt } from 'ethers'
+import { useEffect } from 'react'
+
+const abi = [
+  {
+    type: 'function',
+    name: 'balanceOf',
+    stateMutability: 'view',
+    inputs: [{ name: 'account', type: 'address' }],
+    outputs: [{ type: 'uint256' }],
+  },
+] as const
+
+interface TokenBalanceProps {
+  readonly address: `0x${string}`
+  readonly tokenAddress?: `0x${string}`
+  readonly className?: string
+  readonly toFixed?: number
+  readonly onBalanceChange?: ({
+    balance,
+    formattedBalance,
+  }: {
+    balance: bigint
+    formattedBalance: string | null
+  }) => void
+}
+
+const formatBalance = (balance: bigint, toFixed?: number) => {
+  if (!balance) return null
+  return parseFloat(formatEther(balance, 'wei')).toFixed(toFixed ?? 4)
+}
+
+export const TokenBalance = ({ address, tokenAddress, toFixed, onBalanceChange, className }: TokenBalanceProps) => {
+  const ETHBalance = useBalance({ address })
+  const tokenBalance = useReadContract({
+    abi,
```

Comment:
You could use the default erc20 ABI instead of creating your own. Maybe even create a reusable, default useERC20Read hook, as I believe wagmi removed those in V2 recently.

Reply:
Looks like viem has a erc20 import so i will go with that for now